File: sdk-source/swift/Sources/Aegis/HardwareId.swift
import Foundation
#if canImport(CryptoKit)
import CryptoKit
#endif

/// Stable, non-reversible machine identifier.
public enum HardwareId {
    public static func current() -> String {
        let info = ProcessInfo.processInfo
        let facts = [
            info.hostName,
            info.operatingSystemVersionString,
            info.processorCount.description,
            info.environment["USER"] ?? info.environment["USERNAME"] ?? "",
        ].joined(separator: "|")

        let data = Data(facts.utf8)
        #if canImport(CryptoKit)
        return SHA256.hash(data: data).map { String(format: "%02x", $0) }.joined()
        #else
        var hash: UInt64 = 1469598103934665603
        for byte in data {
            hash = (hash ^ UInt64(byte)) &* 1099511628211
        }
        var out = ""
        var seed = hash
        while out.count < 64 {
            seed = seed &* 6364136223846793005 &+ 1442695040888963407
            out += String(format: "%016lx", seed)
        }
        return String(out.prefix(64))
        #endif
    }
}

File: sdk-source/csharp/Aegis.Sdk/HardwareId.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace Aegis.Sdk
{
    /// <summary>Stable, non-reversible per-machine identifier.</summary>
    public static class HardwareId
    {
        private static string? _cached;

        /// <summary>SHA-256 of machine name + user + OS + processor count.</summary>
        public static string Current
        {
            get
            {
                if (_cached != null) return _cached;
                var seed = string.Join("|",
                    Environment.MachineName,
                    Environment.UserName,
                    Environment.OSVersion.Platform.ToString(),
                    Environment.ProcessorCount.ToString(),
                    Environment.GetEnvironmentVariable("PROCESSOR_IDENTIFIER") ?? "");

                using (var sha = SHA256.Create())
                {
                    var hash = sha.ComputeHash(Encoding.UTF8.GetBytes(seed));
                    var sb = new StringBuilder(64);
                    foreach (var b in hash) sb.Append(b.ToString("x2"));
                    _cached = sb.ToString();
                }

                return _cached;
            }
        }
    }
}